stingray/powerspectrum.py

```diff
@@ -399,3 +399,78 @@ def _make_segment_spectrum(self, lc, segment_size):
             nphots_all.append(np.sum(lc_seg.counts))
 
         return power_all, nphots_all
+
+
+class DynamicalPowerspectrum(AveragedPowerspectrum):
+    def __init__(self, lc, segment_size, norm="frac", gti=None):
+        """
+        Parameters
+        ----------
+        lc : lightcurve.Lightcurve object
+            The time series of which the Dynamical powerspectrum is
+            to be calculated.
+
+        segment_size : float, default 1
+             Length of the segment of light curve, default value is 1 second.
+
+        norm: {"leahy" | "frac" | "abs" | "none" }, optional, default "frac"
+            The normaliation of the periodogram to be used. Options are
+            "leahy", "frac", "abs" and "none", default is "frac".
+
+        Attributes
+        ----------
+        dyn_ps : np.ndarray
+            The matrix with fourier frequencies in the first column and time
+            in the second column.
+        """
+        if segment_size < 2*lc.dt:
+            raise ValueError("Length of the segment is too short to form a "
+                             "light curve!")
+        elif segment_size > lc.tseg:
+            raise ValueError("Length of the segment is too long to create "
+                             "any segments of the light curve!")
+        self.segment_size = segment_size
+        self.norm = norm
+        self.gti = gti
+        self.ps_all, _ = AveragedPowerspectrum._make_segment_spectrum(
+            self, lc, segment_size)
+        self._make_matrix(lc)
+
+    def _make_matrix(self, lc):
+        """Create the matrix with freq and time columns."""
+        row1 = self.ps_all[0].freq
+        row2 = np.arange(lc.time[0] - 0.5*lc.dt + 0.5*self.segment_size,
+                         lc.time[-1] + 0.5*lc.dt, self.segment_size)
+        self.dyn_ps = np.array([row1, row2]).T
+
```

Comment:
Just been reviewing this to see if it's ready for merging, and stumbled over this.

Maybe I'm just confused, but: isn't the actual dynamic PSD stored in `self.ps_all`? Is `self.ps_all` a list of `Powerspectrum` objects? 
So what exactly is in `dyn_ps`? Is that just a list with the frequencies and time steps? If this is the case, I think this might need refactoring. I think `dyn_ps` should store a matrix with the powers, and perhaps we can have two attributes `time` and `freq` that store the time steps and frequencies?

Comments?